src/crimson/common/errorator.h

```diff
@@ -49,24 +49,24 @@ inline auto do_for_each(Container& c, AsyncAction action) {
 
 template<typename AsyncAction>
 inline auto do_until(AsyncAction action) {
```

Comment:
just checked. the comment is still valid.

Reply:
actually, i'd like to drop this helper. but again, this is not in the scope of this change. and i'd defer it to another change if you insist.

Reply:
```diff
diff --git a/src/crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.cc b/src/crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.cc
index ba04440b91e..55e4c9876de 100644
--- a/src/crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.cc
+++ b/src/crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.cc
@@ -191,7 +191,7 @@ LBAInternalNode::find_hole_ret LBAInternalNode::find_hole(
   return seastar::repeat_until_value(
     [i=begin, e=end, c, min_addr, len, this]() mutable {
     if (i == e) {
-      return seastar::make_ready_future<std::optional<laddr_t>>(
+      return find_hole_ertr::make_ready_future<std::optional<laddr_t>>(
         std::make_optional<laddr_t>(L_ADDR_NULL));
     }
     return get_lba_btree_extent(c,
@@ -211,10 +211,7 @@ LBAInternalNode::find_hole_ret LBAInternalNode::find_hole(
       } else {
         return addr;
       }
-    },
-    // TODO: GCC enters a dead loop if crimson::do_until() is used
-    //       or erroratorized future is returned
-    crimson::ct_error::assert_all{ "fix me - APIv6" });
+    });
   });
 }
```

please use this patch to reproduce the issue.

Reply:
I'm not insist.

Reply:
> (I'm not sure because I can still build the errorator test with GCC in both debug and non-debug mode).

@cyx1231st  probably you are still using Seastar_API_LEVEL 5? might want to search `Seastar_API_LEVEL` in your `build/CMakeCache.txt` to verify this.